x-pack/plugins/stack_connectors/public/connector_types/slack_api/slack_connectors.tsx

```diff
@@ -36,20 +42,140 @@ const getSecretsFormSchema = (docLinks: DocLinksStart): SecretsFieldSchema[] =>
   },
 ];
 
-const NO_SCHEMA: ConfigFieldSchema[] = [];
+interface AllowedChannels {
+  id: string;
+  name: string;
+}
+
+const getConfigFormSchemaAfterSecrets = ({
+  options,
+  isLoading,
+  isDisabled,
+  onChange,
+  onCreateOption,
+  selectedOptions,
+}: {
+  options: Array<EuiComboBoxOptionOption<AllowedChannels>>;
+  isLoading: boolean;
+  isDisabled: boolean;
+  onChange: (options: Array<EuiComboBoxOptionOption<AllowedChannels>>) => void;
+  onCreateOption: (searchValue: string, options: EuiComboBoxOptionOption[]) => void;
+  selectedOptions: Array<EuiComboBoxOptionOption<AllowedChannels>>;
+}): ConfigFieldSchema[] => [
+  {
+    id: 'allowedChannels',
+    isRequired: true,
+    label: i18n.ALLOWED_CHANNELS,
+    type: 'COMBO_BOX',
+    euiFieldProps: {
+      noSuggestions: true,
+      isDisabled,
+      isLoading,
+      options,
+      onChange,
+      onCreateOption,
+      selectedOptions,
+    },
+  },
+];
 
+const NO_SCHEMA: ConfigFieldSchema[] = [];
+const SEPARATOR = ' - ';
 export const SlackActionFieldsComponents: React.FC<ActionConnectorFieldsProps> = ({
   readOnly,
   isEdit,
 }) => {
   const { docLinks } = useKibana().services;
 
+  const form = useFormContext();
+  const { setFieldValue } = form;
+  const [formData] = useFormData({ form });
+  const [authToken, setAuthToken] = useState('');
+  const [channelsToValidate, setChannelsToValidate] = useState<string>('');
+  const {
+    channels: validChannels,
+    isLoading,
+    resetChannelsToValidate,
+  } = useValidChannels({
+    authToken,
+    channelId: channelsToValidate,
+  });
+
+  const onCreateOption = useCallback((searchValue: string, options: EuiComboBoxOptionOption[]) => {
+    setChannelsToValidate(searchValue);
+  }, []);
+  const onChange = useCallback(
+    (options: Array<EuiComboBoxOptionOption<AllowedChannels>>) => {
+      const tempChannelIds: AllowedChannels[] = options.map(
+        (opt: EuiComboBoxOptionOption<AllowedChannels>) => {
+          return opt.value!;
+        }
+      );
+      resetChannelsToValidate(tempChannelIds);
+    },
+    [resetChannelsToValidate]
+  );
+
+  const configFormSchemaAfterSecrets = useMemo(() => {
+    const validChannelsFormatted = validChannels.map((channel) => ({
+      label: `${channel.id}${SEPARATOR}${channel.name}`,
+      value: channel,
```

Comment:
nit: is it more user friendly to put the channel name first?

Reply:
I think it is always better to put first what the user type, so they do not get confused